Tested locally gpertools, I can confirm the bug. When using master branch, and building to ios, I have the following error:

```
configure: error: cannot run C compiled programs.
If you meant to cross compile, use '--host'.
```

Full build log: [gperftools-2.17.2-ios-armv8-clang21-release.log](https://github.com/user-attachments/files/28799993/gperftools-2.17.2-ios-armv8-clang21-release.log)

Once patched, it works properly: [gperftools-2.17.2-ios-armv8-clang21-release-patched.log](https://github.com/user-attachments/files/28800034/gperftools-2.17.2-ios-armv8-clang21-release-patched.log)

-----

libuuid has the same scenario, fails on master, but works after the patch:

* On master, fails just like gperftools: [libuuid-ios-armv8-clang21-release.log](https://github.com/user-attachments/files/28800664/libuuid-ios-armv8-clang21-release.log)

* Once patched, works fine: [libuuid-ios-armv8-clang21-release-patched.log](https://github.com/user-attachments/files/28800662/libuuid-ios-armv8-clang21-release-patched.log)

--- a/recipes/gperftools/all/conanfile.py
+++ b/recipes/gperftools/all/conanfile.py
@@ -1,6 +1,6 @@
 from conan import ConanFile
 from conan.errors import ConanInvalidConfiguration
-from conan.tools.apple import fix_apple_shared_install_name, is_apple_os, XCRun
+from conan.tools.apple import fix_apple_shared_install_name
 from conan.tools.build import cross_building, check_min_cppstd, stdcpp_library
 from conan.tools.cmake import cmake_layout
 from conan.tools.env import VirtualRunEnv

--- a/recipes/util-linux-libuuid/all/conanfile.py
+++ b/recipes/util-linux-libuuid/all/conanfile.py
@@ -1,6 +1,6 @@
 from conan import ConanFile
 from conan.errors import ConanInvalidConfiguration
-from conan.tools.apple import fix_apple_shared_install_name, is_apple_os, XCRun
+from conan.tools.apple import fix_apple_shared_install_name
 from conan.tools.files import copy, get, rm, rmdir
 from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
 from conan.tools.layout import basic_layout

